src/oop/user_defined_classes/hard/q5/MainTest.java

```diff
@@ -0,0 +1,109 @@
+        assertTrue(classInstance.hasField(doubleAttributeName2, double.class),
+                String.format(wrongTypeMessage, doubleAttributeName2));
+        assertTrue(classInstance.hasModifier(doubleAttributeName2, "private"),
+                String.format(incorrectModifierMessage, doubleAttributeName2, testClassName));
+    }
+
+    @Test
+    public void phoneClassHasRequiredConstructor() {
+        Class<?>[] arguments = {
+                double.class, double.class
+        };
+        assertTrue(classInstance.hasConstructor(arguments),
+                String.format("Your %s class is missing a required constructor.", testClassName));
+        assertTrue(classInstance.hasConstructor(arguments, new String[]{"public"}),
+                String.format("Your %s class constructor has the incorrect visibility modifier.", testClassName));
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(0.0, 1.2),
+                Arguments.of(5.42, 11.11),
+                Arguments.of(589.6, 432.9)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void phoneConstructorInitializesAttributesCorrectly(double attribute1, double attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, double.class},
+                {attribute2, double.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeInstantiationMessage =
+                "Your " + testClassName + " constructor does not correctly initialize the %s attribute.";
+        _assertEquals(attribute1, classInstance.getFieldValue(instance, doubleAttributeName1),
+                String.format(incorrectAttributeInstantiationMessage, doubleAttributeName1));
+        _assertEquals(attribute2, classInstance.getFieldValue(instance, doubleAttributeName2),
+                String.format(incorrectAttributeInstantiationMessage, doubleAttributeName2));
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(1, 0),
+                Arguments.of(0, 2),
+                Arguments.of(23.6, 8.1),
+                Arguments.of(174.429, 482.4893)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void correctCalcScreenSizeMethod(double attribute1, double attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, double.class},
+                {attribute2, double.class}
+        };
+        double diag = Math.sqrt(Math.pow(attribute1, 2) + Math.pow(attribute2, 2));
+        Object instance = classInstance.createInstance(arguments);
+        assertTrue(classInstance.hasMethod(methodName, null, double.class, new String[]{"public"}),
```

Comment:
I would check the return type of the method separately so we can give the student better feedback.